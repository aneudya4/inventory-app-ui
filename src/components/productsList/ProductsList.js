import React from 'react';
import Product from '../product/Product';
import { Link } from 'react-router-dom';
import config from '../config';

const ProductsList = ({ products, updateProducts, addToCart }) => {
  return (
    <div className='product-list'>
      <ul>
        {products.map((product) => {
          return (
            <li key={product.id}>
              <Product
                product={product}
                updateProducts={updateProducts}
                addToCart={addToCart}
              />
            </li>
          );
        })}
      </ul>
    </div>
  );
};

export default ProductsList;
