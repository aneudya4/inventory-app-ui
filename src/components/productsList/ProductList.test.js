import { shallow } from 'enzyme';

import ProductsList from './ProductsList.js';

describe('Test on the <ProductList/> Component', () => {
  it('renders ProductList without crashing', () => {
    const testProduct = [
      {
        id: 1,
        product_name: 'Iphone 12 Pro',
        user_id: 1,
        unit_price: '799.99',
        description: '64GB',
        stock_total: 18,
        provider: 'Apple',
      },
    ];

    const wrapper = shallow(<ProductsList products={testProduct} />);

    expect(wrapper).toMatchSnapshot();
  });
});
