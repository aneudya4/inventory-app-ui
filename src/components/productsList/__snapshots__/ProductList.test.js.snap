// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on the <ProductList/> Component renders ProductList without crashing 1`] = `
<div
  className="product-list"
>
  <h3>
    Products
  </h3>
  <ul>
    <withRouter(Product)
      key="1"
      product={
        Object {
          "description": "64GB",
          "id": 1,
          "product_name": "Iphone 12 Pro",
          "provider": "Apple",
          "stock_total": 18,
          "unit_price": "799.99",
          "user_id": 1,
        }
      }
    />
  </ul>
</div>
`;
